src/main/scala/chisel3/util/Math.scala

```diff
@@ -8,7 +8,21 @@ package chisel3.util
 import chisel3._
 import chisel3.internal.chiselRuntimeDeprecated
 
-/** Compute the log2 rounded up with min value of 1 */
+/** Compute the log2 of a Scala integer, rounded up, with min value of 1.
+  * Useful for getting the number of bits needed to represent some number of states (in - 1),
+  * To get the number of bits needed to represent some number n, use log2Up(n + 1).
+  * with the minimum value preventing the creation of currently-unsupported zero-width wires.
+  *
+  * Note: prefer to use log2Ceil when in is known to be > 1 (where log2Ceil(in) > 0).
+  * This will be deprecated when zero-width wires is supported.
+  *
+  * @example {{{
+  * log2Up(1) = 1
```

Comment:
Should the example be executable? Maybe something like `assert(log2Up(1) == 1)`

Reply:
Do we have stylistic conventions on that? That's probably good to figure out.

Reply:
poke?

Reply:
Fixed. Didn't realize this was a blocking issue. I opted to put the return value in comments because the assert is not part of the example.